server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -317,8 +326,16 @@ public void childEvent(
     catch (Exception ex) {
       throw ex;
     }
-    Assert.assertTrue(assignSegmentLatch.await(4, TimeUnit.SECONDS));
+
+    Assert.assertTrue(assignSegmentLatch.await(2, TimeUnit.SECONDS));
+    Assert.assertEquals(ImmutableMap.of(dataSource, 100.0), coordinator.getLoadStatus());
     curator.delete().guaranteed().forPath(ZKPaths.makePath(LOADPATH, dataSegment.getIdentifier()));
+    // Wait for coordinator thread to run so that replication status is updated
+    // Waiting 50 ms more than the Coordination period should be sufficient
+    Thread.sleep(COORDINATOR_PERIOD + 50);
```

Comment:
Instead of sleep, can check for the test condition in a while loop and put a test timeout outside this test method. However, the while loop without sleep inside it will be busy waiting for the condition to be satisfied.